community/kernel/src/main/java/org/neo4j/helpers/ArrayUtil.java

```diff
@@ -485,6 +486,27 @@ public static boolean isEmpty( Object[] array )
         return length == result.length ? result : Arrays.copyOf( result, length );
     }
 
+    /**
+     * Filters an array, only including the items that passes the {@code filter}.
+     *
+     * @param array items to filter.
+     * @param filter {@link Predicate} to match each item.
+     * @return a new array with the items that passed the filter.
+     */
+    public static <T> T[] filter( T[] array, Predicate<T> filter )
```

Comment:
In both cases where this method is used same can be achieved using `Stream.of` or `Arrays.stream` 
using plain stream api.